pubsub/benchmarks_test.go

```diff
@@ -0,0 +1,120 @@
+// Copyright 2019 The Go Cloud Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package pubsub_test
+
+import (
+	"context"
+	"flag"
+	"testing"
+
+	"gocloud.dev/gcp"
+	"gocloud.dev/pubsub"
+	"gocloud.dev/pubsub/gcppubsub"
+	"golang.org/x/sync/errgroup"
+	"golang.org/x/sync/semaphore"
+)
+
+var (
+	projectID = flag.String("benchmark-project", "", "project ID")
+	topicName = flag.String("benchmark-topic", "", "topic name")
+)
+
+func BenchmarkReceive(b *testing.B) {
```

Comment:
Why not do this as a conformance benchmark like the one I added for `blob`?
https://github.com/google/go-cloud/blob/master/blob/drivertest/drivertest.go#L202

Reply:
That's a good idea. I was focussing on GCP performance now because I've been comparing us to cloud.google.com/go/pubsub. But I will do that in a followup PR.

Reply:
#1147